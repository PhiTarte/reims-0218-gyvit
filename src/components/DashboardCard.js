import React from "react"
import {
  Card,
  CardImg,
  CardText,
  CardBody,
  CardTitle,
  Button,
  Container,
  Row
} from "reactstrap"

const DashboardCard = ({ contenu }) => {
  const { image, titre, bio, buttonText, link } = { ...contenu }
  return (
    <div>
      <Container>
        <Row>
          <Card
            style={{
              width: "288px",
              height: "423px",
              "margin-right": "32px",
              padding: "0px"
            }}
          >
            <CardImg top width="100%" src={image} alt="Card image cap" />
            <CardBody>
              <CardTitle style={{ size: "18px", "text-align": "center" }}>
                {titre}
              </CardTitle>
              <CardText
                style={{
                  width: "228px",
                  height: "96px",
                  size: "15px",
                  opacity: 0.45,
                  "text-align": "center"
                }}
              >
                {bio}
              </CardText>
              <Button
                color="link"
                style={{
                  width: "170px",
                  height: "44px",
                  size: "16px",
                  "margin-right": "59px",
                  "margin-left": "59px ",
                  "border-radius": "2px",
                  padding: "0px"
                }}
                src={link}
              >
                {buttonText}
              </Button>
            </CardBody>
          </Card>
        </Row>
      </Container>
    </div>
  )
}
export default DashboardCard
