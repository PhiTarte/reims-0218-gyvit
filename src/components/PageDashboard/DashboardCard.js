import React from "react"
import {
  Card,
  CardImg,
  CardText,
  CardBody,
  CardTitle,
  Button,
  Container,
  Row
} from "reactstrap"
import PropTypes from "prop-types"

const DashboardCard = ({ contenu }) => {
  const { titre, bio, buttonText, link } = { ...contenu }
  return (
    <div>
      <Container>
        <Row>
          <Card
            style={{
              width: "288px",
              height: "423px",
              "margin-right": "32px"
            }}
          >
            <CardImg
              src={process.env.PUBLIC_URL + "img/folder.ico"}
              alt="folder"
              style={{
                width: "151.2px",
                height: "120px",
                "margin-top": "31px",
                "margin-left": "72px",
                "margin-right": "64.8px"
              }}
            />
            <CardBody
              style={{
                padding: "0px"
              }}
            >
              <CardTitle
                style={{
                  width: "175px",
                  height: "24px",
                  "font-size": "18px",
                  "text-align": "center",
                  "padding-left": "0px",
                  "padding-right": "0px",
                  "margin-top": "15px",
                  "margin-left": "57px",
                  "margin-right": "56px"
                }}
              >
                {titre}
              </CardTitle>
              <CardText
                style={{
                  width: "228px",
                  height: "96px",
                  "font-size": "15px",
                  opacity: 0.45,
                  "text-align": "center",
                  "margin-top": "13px",
                  "margin-left": "30px",
                  "margin-right": "30px",
                  "margin-bottom": "50px",
                  "line-height": "24px"
                }}
              >
                {bio}
              </CardText>
              <Button
                style={{
                  width: "170px",
                  height: "44px",
                  "font-size": "16px",
                  "border-radius": "2px",
                  "margin-right": "59px",
                  "margin-left": "59px ",
                  "margin-bottom": "30px",
                  padding: "0px"
                }}
                src={link}
              >
                {buttonText}
              </Button>
            </CardBody>
          </Card>
        </Row>
      </Container>
    </div>
  )
}

DashboardCard.propTypes = {
  contenu: {
    id: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
    titre: PropTypes.string.isRequired,
    bio: PropTypes.string.isRequired,
    buttonText: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired
  }
}

export default DashboardCard
