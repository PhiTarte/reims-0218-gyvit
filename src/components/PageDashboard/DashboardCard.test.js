import React from "react"
import ReactDOM from "react-dom"
import DashboardCard from "./DashboardCard"
import renderer from "react-test-renderer"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<DashboardCard />, div)
  ReactDOM.unmountComponentAtNode(div)
})
it("renders link html", () => {
  const tree = renderer.create(<DashboardCard />).toJSON()
  expect(tree).toMatchSnapshot()
})
