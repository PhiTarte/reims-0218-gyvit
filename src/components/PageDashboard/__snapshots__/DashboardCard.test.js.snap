// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link html 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="card"
        style={
          Object {
            "height": "423px",
            "marginRight": "32px",
            "width": "288px",
          }
        }
      >
        <img
          alt="folder"
          className="card-img"
          src="img/folder.ico"
          style={
            Object {
              "height": "120px",
              "marginLeft": "72px",
              "marginRight": "64.8px",
              "marginTop": "31px",
              "width": "151.2px",
            }
          }
        />
        <div
          className="card-body"
          style={
            Object {
              "padding": "0px",
            }
          }
        >
          <h5
            className="card-title"
            style={
              Object {
                "fontSize": "18px",
                "height": "24px",
                "marginLeft": "57px",
                "marginRight": "56px",
                "marginTop": "15px",
                "paddingLeft": "0px",
                "paddingRight": "0px",
                "textAlign": "center",
                "width": "175px",
              }
            }
          />
          <p
            className="card-text"
            style={
              Object {
                "fontSize": "15px",
                "height": "96px",
                "lineHeight": "24px",
                "marginBottom": "50px",
                "marginLeft": "30px",
                "marginRight": "30px",
                "marginTop": "13px",
                "opacity": 0.45,
                "textAlign": "center",
                "width": "228px",
              }
            }
          />
          <button
            className="btn btn-secondary"
            onClick={[Function]}
            src={undefined}
            style={
              Object {
                "borderRadius": "2px",
                "fontSize": "16px",
                "height": "44px",
                "marginBottom": "30px",
                "marginLeft": "59px ",
                "marginRight": "59px",
                "padding": "0px",
                "width": "170px",
              }
            }
            type={undefined}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
